crates/autopilot/src/domain/auction/order.rs

```diff
@@ -0,0 +1,124 @@
+use primitive_types::{H160, H256, U256};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Order {
+    pub uid: OrderUid,
+    pub sell_token: H160,
+    pub buy_token: H160,
+    pub sell_amount: U256,
+    pub buy_amount: U256,
+    pub solver_fee: U256,
+    pub user_fee: U256,
+    pub kind: Kind,
+    pub class: Class,
+    pub valid_to: u32,
+    pub receiver: Option<H160>,
+    pub owner: H160,
+    pub partially_fillable: bool,
+    pub executed: U256,
+    // Partially fillable orders should have their pre-interactions only executed
+    // on the first fill.
+    pub pre_interactions: Vec<Interaction>,
+    pub post_interactions: Vec<Interaction>,
+    pub sell_token_balance: SellTokenSource,
+    pub buy_token_balance: BuyTokenDestination,
+    pub app_data: AppDataHash,
+    pub signature: Signature,
+}
+
+// uid as 56 bytes: 32 for orderDigest, 20 for ownerAddress and 4 for validTo
+#[derive(Copy, Clone, Debug, PartialEq, Hash, Eq)]
+pub struct OrderUid(pub [u8; 56]);
+
+impl Order {
+    pub fn is_limit_order(&self) -> bool {
+        matches!(self.class, Class::Limit)
+    }
+
+    /// For some orders the protocol doesn't precompute a fee. Instead solvers
+    /// are supposed to compute a reasonable fee themselves.
+    pub fn solver_determines_fee(&self) -> bool {
+        self.is_limit_order()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Kind {
+    Buy,
+    Sell,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Class {
+    Market,
+    Limit,
+    Liquidity,
```

Comment:
One nice side effect of separating the data models is that in the autopilot we can get rid of Liquidity orders (they should just be treated as limit orders). Not for this PR, but another reason I think this is going in the right direction.